src/query/ast/src/ast/statements/copy.rs

```diff
@@ -189,3 +189,33 @@ impl Display for UriLocation {
         Ok(())
     }
 }
+
+pub enum CopyOptionItem {
+    Files(Vec<String>),
+    Pattern(String),
+    FileFormat(BTreeMap<String, String>),
+    ValidationMode(String),
+    SizeLimit(usize),
+    MaxFileSize(usize),
+    SplitSize(usize),
+    Single(bool),
+    Purge(bool),
+    Force(bool),
+}
+
+impl CopyOptionItem {
+    pub fn apply(&self, stmt: &mut CopyStmt<'_>) {
```

Comment:
How about impl CopyStmt with `pub fn apply_option(&mut self, opt: CopyOption)`? We can move and consume the CopyOption directly without an extra clone.

Reply:
yep, lemme change it to move